system-contracts/contracts/precompiles/EcAdd.yul

```diff
@@ -413,28 +441,35 @@ object "EcAdd" {
 
             // P1 + P2 = P3
 
-            x1 := intoMontgomeryForm(x1)
-            y1 := intoMontgomeryForm(y1)
-            x2 := intoMontgomeryForm(x2)
-            y2 := intoMontgomeryForm(y2)
+            m_x1 := intoMontgomeryForm(x1)
+            m_y1 := intoMontgomeryForm(y1)
+            m_x2 := intoMontgomeryForm(x2)
+            m_y2 := intoMontgomeryForm(y2)
 
             // Ensure that the points are in the curve (Y^2 = X^3 + 3).
-            if or(iszero(pointIsInCurve(x1, y1)), iszero(pointIsInCurve(x2, y2))) {
+            if or(iszero(pointIsInCurve(m_x1, m_y1)), iszero(pointIsInCurve(m_x2, m_y2))) {
                 burnGas()
             }
 
-            // (y2 - y1) / (x2 - x1)
-            let slope := montgomeryDiv(submod(y2, y1, P()), submod(x2, x1, P()))
-            // x3 = slope^2 - x1 - x2
-            let x3 := submod(montgomeryMul(slope, slope), addmod(x1, x2, P()), P())
-            // y3 = slope * (x1 - x3) - y1
-            let y3 := submod(montgomeryMul(slope, submod(x1, x3, P())), y1, P())
-
-            x3 := outOfMontgomeryForm(x3)
-            y3 := outOfMontgomeryForm(y3)
+            mstore(0, x1)
+            mstore(32, y1)
+            mstore(64, x2)
+            mstore(96, y2)
+
+            let precompileParams := unsafePackPrecompileParams(
+                0, // input offset in words
+                4, // input length in words (x1, y1, x2, y2)
+                0, // output offset in words
+                2, // output length in words (x, y)
+                0  // No special meaning, ecadd circuit doesn't check this value
+            )
+            let gasToPay := ECADD_GAS_COST()
+
+            let success := precompileCall(precompileParams, gasToPay)
```

Comment:
You should remove all unused code.

Reply:
Hi, @vladbochok, the code I have left is used to check that the point is on the curve as well as coordinates are less that modulo

Reply:
Right, I have just though that it is better to not modify audited code as it serves its purpose. Additionally, there is premature handling of corner-case operations involving Montgomery form, such as Inf + 0 etc. In-circuit we bring only after passing all the corner-cases and checks.

Reply:
Of course, I would however try to leverage in-circuit checks, leaving a blank call to it from yul code to achieve end-to-end best performance and security